saleor/graphql/account/mutations/jwt.py

```diff
@@ -0,0 +1,258 @@
+            "Invalid token", code=AccountErrorCode.JWT_INVALID_TOKEN.value
+        )
+    return payload
+
+
+def get_user(payload):
+    user = get_user_from_payload(payload)
+    if not user:
+        raise ValidationError(
+            "Invalid token", code=AccountErrorCode.JWT_INVALID_TOKEN.value
+        )
+    return user
+
+
+class CreateToken(BaseMutation):
+    """Mutation that authenticates a user and returns token and user data."""
+
+    class Arguments:
+        email = graphene.String(required=True, description="Email of a user.")
+        password = graphene.String(required=True, description="Password of a user.")
+
+    class Meta:
+        description = "Create JWT token."
+        error_type_class = AccountError
+        error_type_field = "account_errors"
+
+    token = graphene.String(description="JWT token, required to authenticate.")
+    refresh_token = graphene.String(
+        description="JWT refresh token, required to re-generate access token."
+    )
+    csrf_token = graphene.String(
+        description="CSRF token required to re-generate access token."
+    )
+    user = graphene.Field(User, description="A user instance.")
+
+    @classmethod
+    def get_user(cls, info, data):
+        user = authenticate(
+            request=info.context, username=data["email"], password=data["password"],
+        )
+        if not user:
+            raise ValidationError(
+                {
+                    "email": ValidationError(
+                        "Please, enter valid credentials",
+                        code=AccountErrorCode.INVALID_CREDENTIALS.value,
+                    )
+                }
+            )
+        return user
+
+    @classmethod
+    def perform_mutation(cls, root, info, **data):
+        user = cls.get_user(info, data)
+        access_token = create_access_token(user)
+        csrf_token = _get_new_csrf_token()
+        refresh_token = create_refresh_token(user, {"csrfToken": csrf_token})
+        info.context.refresh_token = refresh_token
+        info.context._cached_user = user
+        user.last_login = timezone.now()
+        user.save(update_fields=["last_login"])
+        return cls(
+            errors=[],
+            user=user,
+            token=access_token,
+            refresh_token=refresh_token,
+            csrf_token=csrf_token,
+        )
+
+
+class RefreshToken(BaseMutation):
+    """Mutation that refresh user token and returns token and user data."""
+
+    token = graphene.String(description="JWT token, required to authenticate.")
+    user = graphene.Field(User, description="A user instance.")
+
+    class Arguments:
+        refresh_token = graphene.String(required=False, description="Refresh token.")
+        csrf_token = graphene.String(
+            required=True, description="CSRF token required to refresh token."
+        )
+
+    class Meta:
+        description = (
+            "Refresh JWT token. Mutation tries to take refreshToken from "
```

Comment:
This is incorrect, it documents the previous behavior.

Reply:
Thanks for catching this.